### ISSUE TITLE
SAML Request signing broken due to `strip!` method

### ISSUE THREAD

**Issue by bramleyjl** [SAML-Toolkits/ruby-saml#643]
We are updating our SAML authentication requests to our service provider by including SAML signing, however the resulting `DigestValue` is always blank. I traced the problem back to the [`compute_digest` method](https://github.com/onelogin/ruby-saml/blob/master/lib/xml_security.rb#L178), which is using [`strip!`](https://apidock.com/ruby/v2_5_5/String/strip%21) to remove preceding and trailing whitespaces.

```
    def compute_digest(document, digest_algorithm)
      digest = digest_algorithm.digest(document)
      Base64.encode64(digest).strip!
    end
```

We are using `OpenSSL::Digest::SHA256` as our `digest_algorithm`, but the Base64 encoding of the digest created on the previous line doesn't contain any preceding or trailing whitespaces, causing the `strip!` method to return `nil` instead of the supplied value. We have also attempted to alter the SAML metadata provided to the ruby-saml gem in order to introduce a whitespace to pass this method but were unable to find a way to do so.

What is the purpose of using `strip!` over the base [`strip`](https://apidock.com/ruby/v2_5_5/String/strip) method? Is it possible for it to be replaced with `strip` in this instance?

**Comment by StoneCypher** [comment]
OneLogin stopped maintaining this library more than a month ago

https://github.com/onelogin/ruby-saml/issues/640

**Comment by StoneCypher** [comment]
@pitbulk was trying to arrange takeover rights.  I don't know whether that ended up happening

**Comment by mrmoss** [comment]
Still working on getting it swapped over. @eriktalvi - Any updates?

**Comment by mrmoss** [comment]
@pitbulk - You want to just start forking things? I don't have much faith in the slow bureaucracy that is OL these days. The team will continue to pursue getting these transferred over, but I'd rather put the security of users first personally...something something actually trying to live those LPs...

Let us know how we can help. Could assign @eriktalvi to help with that forking process as long as @Subterrane approves.

**Comment by pitbulk** [comment]
I had a conversation today with Tat Ng (VP Engineering) and it was cool, I hope to have such access soon.

**Comment by mrmoss** [comment]
Nice. Maybe don't fork just yet then :)

**Comment by pitbulk** [comment]
Fixed at #650
